schema/schemaspec/schemahcl/hcl_test.go

```diff
@@ -137,3 +138,10 @@ func ExampleMarshal() {
 	//   y = 1
 	// }
 }
+
+func TestInterface(t *testing.T) {
+	var (
+		_ schemaspec.Unmarshaler = &defaultCodec{}
+		_ schemaspec.Marshaler   = &defaultCodec{}
+	)
+}
```

Comment:
This can be replaced with a compile-time check.

```go
var _ interface {
	schemaspec. Marshaler
	schemaspec.Unmarshaler
} = (*defaultCodec)(nil)
```